test/releasers/terraform-module.ts

```diff
@@ -0,0 +1,154 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {describe, it, afterEach, beforeEach} from 'mocha';
+import {TerraformModule} from '../../src/releasers/terraform-module';
+import {readFileSync} from 'fs';
+import {resolve} from 'path';
+import {stringifyExpectedChanges, readPOJO} from '../helpers';
+import * as nock from 'nock';
+import * as snapshot from 'snap-shot-it';
+import * as suggester from 'code-suggester';
+import * as sinon from 'sinon';
+
+nock.disableNetConnect();
+const sandbox = sinon.createSandbox();
+const fixturesPath = './test/releasers/fixtures/terraform';
+const releasePR = new TerraformModule({
+  repoUrl: 'googleapis/terraform-test-repo',
+  releaseType: 'terraform-module',
+  // not actually used by this type of repo.
+  packageName: 'terraform-test-repo',
+  apiUrl: 'https://api.github.com',
+});
+
+describe('terraform-module', () => {
+  const tests = [
+    {
+      // simple-module with module versions defined
+      name: 'simple-module',
+      findVersionFiles: ['versions.tf'],
+      readFilePaths: ['simple-module/readme.md', 'simple-module/versions.tf'],
+    },
+    {
+      // module-submodule with submodules
+      // sub-module-with-version has module versions defined
+      // sub-module-missing-versions has no versions.tf
+      name: 'module-submodule',
+      findVersionFiles: [
+        'versions.tf',
+        'modules/sub-module-with-version/versions.tf',
+      ],
+      readFilePaths: [
+        'module-submodule/readme.md',
+        'module-submodule/versions.tf',
+        'module-submodule/modules/sub-module-with-version/versions.tf',
+      ],
+    },
+    {
+      // module-no-versions with no module versions defined in versions.tf
+      name: 'module-no-versions',
+      findVersionFiles: [],
+      readFilePaths: ['module-no-versions/readme.md'],
+    },
+  ];
+  beforeEach(() => {
+    // Indicates that there are no PRs currently waiting to be released:
+    sandbox
+      .stub(releasePR.gh, 'findMergedReleasePR')
+      .returns(Promise.resolve(undefined));
+    // Commits, used to build CHANGELOG, and propose next version bump:
+    sandbox
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      .stub(releasePR as any, 'commits')
+      .returns(Promise.resolve(readPOJO('commits-fix')));
+    // See if there are any release PRs already open, we do this as
+    // we consider opening a new release-pr:
+    sandbox
+      .stub(releasePR.gh, 'findOpenReleasePRs')
+      .returns(Promise.resolve([]));
+
+    // Lookup the default branch name:
+    sandbox.stub(releasePR.gh, 'getDefaultBranch').resolves('main');
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+  describe('run', () => {
```

Comment:
These tests seem really solid to me, do you think it was easier to mock the methods in `github` rather than trying to mock the routes with `nock`?

Reply:
I have never used `nock` but the simple-module was a great starting point for me with Sinon.